File: PhysioHealthCare.Domain/Entities/Patient.cs
namespace PhysioHealthCare.Domain.Entities
{
    using PhysioHealthCare.Domain.Common;
    using PhysioHealthCare.Domain.Enums;
    public class Patient : BaseEntity
    {
        public string FirstName { get; set; } = string.Empty;

        public string LastName { get; set; } = string.Empty;

        public DateTime BirthDate { get; set; }

        public Gender Gender { get; set; }

        public string PhoneNumber { get; set; } = string.Empty;

        public string Email { get; set; } = string.Empty;

        public string Address { get; set; } = string.Empty;

        public string Notes { get; set; } = string.Empty;
    }
}

File: PhysioHealthCare.Web/src/app/shared/models/patient.ts
export type PatientGender =
  | 'Male'
  | 'Female'
  | 'Other';

export type PatientGenderValue =
  | 1
  | 2
  | 3;

export type PatientSortField =
  | 'fullName'
  | 'birthDate'
  | 'gender'
  | 'phoneNumber'
  | 'email';

export type SortDirection =
  | 'asc'
  | 'desc';


export interface Patient {
  id: string;

  fullName: string;

  birthDate: string;

  gender: PatientGender;

  phoneNumber: string;

  email: string;
}


export interface CreatePatient {
  firstName: string;

  lastName: string;

  birthDate: string;

  gender: PatientGenderValue;

  phoneNumber?: string;

  email?: string;

  address?: string;

  notes?: string;
}


export interface UpdatePatient {
  firstName: string;

  lastName: string;

  birthDate: string;

  gender: PatientGenderValue;

  phoneNumber?: string;

  email?: string;

  address?: string;

  notes?: string;
}


export interface PatientDetail {
  id: string;

  firstName: string;

  lastName: string;

  birthDate: string;

  gender: PatientGenderValue;

  phoneNumber: string | null;

  email: string | null;

  address: string | null;

  notes: string | null;
}